src/stdlib.c

```diff
@@ -231,26 +231,38 @@ static void init_js_stdlib(struct v7 *v7) {
   val_t res;
 
   v7_exec(v7, &res, STRINGIFY(
-Array.prototype.indexOf = function(a, b) {
-  if (!b || b < 0) b = 0;
-  for (var i = b; i < this.length; i++) {
-    if (this[i] == a) {
-      return i;
-    }
-  }
-  return -1;
-};
-
-Array.prototype.lastIndexOf = function(a, b) {
-  if (!b || b < 0 || b >= this.length) b = this.length - 1;
-  for (var i = b; i >= 0; i--) {
-    if (this[i] == a) {
-      return i;
-    }
-  }
-  return -1;
-};
-));
+    Array.prototype.indexOf = function(a, b) {
+      if (!b || b < 0) b = 0;
+      for (var i = b; i < this.length; i++) {
+        if (this[i] == a) {
+          return i;
+        }
+      }
+      return -1;
+    };));
+
+  v7_exec(v7, &res, STRINGIFY(
+    Array.prototype.lastIndexOf = function(a, b) {
+      if (!b || b < 0 || b >= this.length) b = this.length - 1;
+      for (var i = b; i >= 0; i--) {
+        if (this[i] == a) {
+          return i;
+        }
+      }
+      return -1;
+    };
+    ));
+
+  v7_exec(v7, &res, STRINGIFY(
+    Array.prototype.pop = function(a, b) {
+      var i = this.length - 1;
+      return this.splice(i, 1)[0];
+    };));
+
+  v7_exec(v7, &res, STRINGIFY(
+    Array.prototype.shift = function(a, b) {
```

Comment:
Ah noticed now: spurious arguments (probably copy pasta).
There are actually tests that check `Array.prototype.pop.length` (which didn't work anyway until PR #320)